### ISSUE TITLE
ValueError in loading Object arrays in the response for SKlearnPredictor

### ISSUE THREAD

**Issue by sai-guruju** [aws/sagemaker-python-sdk#1753]
**Bug Description**
Encountered an error when calling SKLearnPredictor.predict for numpy >= 1.17.4. Code responded with "ValueError: Object arrays cannot be loaded when allow_pickle=False", also SKLLearnPredictor doesn't accept a custom deserializer. 

**To reproduce**
`import sagemaker`
`from boto3.session import Session`
`from sagemaker.sklearn.model import SKLearnPredictor`
`sagemaker_session = sagemaker.Session(`
`    boto_session=Session(aws_access_key_id='access_key',`
`                         aws_secret_access_key='secret_key',`
`                         region_name='region_name')`
`)`
`svm_predictor = SKLearnPredictor(`
`    endpoint_name='endpoint_name',`
`    sagemaker_session=sagemaker_session`
`)`
`test_case = [{'text_a': '', 'text_b': ''}] * 2`
`a = svm_predictor.predict(test_case)`
`print(a) # this is a numpy object array`

**Expected behavior**
SKLearnPredictor should take a custom deserializer or a parameter for allow_pickle for handling deserialisation of sagemaker endpoint response.  

**Screenshots or logs**
`Traceback (most recent call last):`
`  File "testing.py", line 74, in <module>`
`    a = svm_predictor.predict(test_case)`
`  File "/Users/some_path/python3.7/site-packages/sagemaker/predictor.py", line 114, in predict`
`    return self._handle_response(response)`
`  File "/Users/some_path/python3.7/site-packages/sagemaker/predictor.py", line 124, in _handle_response`
`    return self.deserializer(response_body, response["ContentType"])`
`  File "/Users/some_path/python3.7/site-packages/sagemaker/predictor.py", line 626, in __call__`
`    return np.load(BytesIO(stream.read()))`
`  File "/Users/some_path/python3.7/site-packages/numpy/lib/npyio.py", line 453, in load`
`    pickle_kwargs=pickle_kwargs)`
`  File "/Users/some_path/python3.7/site-packages/numpy/lib/format.py", line 722, in read_array`
`    raise ValueError("Object arrays cannot be loaded when "`
`ValueError: Object arrays cannot be loaded when allow_pickle=False`

**System information**
- **SageMaker Python SDK version**: 1.69.0
- **scikit-learn**: 0.20.0
- **Python version**: 3.7.4

**Comment by laurenyu** [comment]
it is possible to provide a custom deserializer, though admittedly it's not very clean (we have some upcoming work planned around that):

```python
svm_predictor = SKLearnPredictor(...)
svm_predictor.deserializer = my_custom_deserializer
```

as for the actual error message with numpy, I'll chat with the team and see if we want to make `allow_pickle=True` to be the default behavior.

**Comment by laurenyu** [comment]
fixed in #1755. This will get released as part of v2.0.0 (hopefully this week or next)

**Comment by laurenyu** [comment]
released as part of [v2.0.0](https://github.com/aws/sagemaker-python-sdk/releases/tag/v2.0.0)
